packages/remix-eslint-config/rules/typescript.js

```diff
@@ -1,54 +1,49 @@
-const OFF = 0;
-const WARN = 1;
-const ERROR = 2;
-
 module.exports = {
-  "no-dupe-class-members": OFF,
-  "no-undef": OFF,
-
-  // Add TypeScript specific rules (and turn off ESLint equivalents)
-  "@typescript-eslint/consistent-type-assertions": WARN,
-  "@typescript-eslint/consistent-type-imports": WARN,
-
-  "no-array-constructor": OFF,
-  "@typescript-eslint/no-array-constructor": WARN,
+  // TODO: These rules might be nice to enable... we should investigate eventually!
+  "@typescript-eslint/ban-ts-comment": "off",
+  "@typescript-eslint/ban-types": "off",
+  "@typescript-eslint/no-empty-function": "off",
+  "@typescript-eslint/no-empty-interface": "off",
+  "@typescript-eslint/no-explicit-any": "off",
+  "@typescript-eslint/no-inferrable-types": "off",
+  "@typescript-eslint/no-namespace": "off",
+  "@typescript-eslint/no-non-null-assertion": "off",
+  "@typescript-eslint/no-var-requires": "off",
+  "no-var": "off",
+  "prefer-rest-params": "off",
 
-  // There is a bug w/ @typescript-eslint/no-duplicate-imports triggered
-  // by multiple imports inside of module declarations. We should reenable
-  // this rule when the bug is fixed.
-  // https://github.com/typescript-eslint/typescript-eslint/issues/3071
-  "no-duplicate-imports": OFF,
-  // "@typescript-eslint/no-duplicate-imports": WARN,
-
-  "no-redeclare": OFF,
-  "@typescript-eslint/no-redeclare": ERROR,
-  "no-use-before-define": OFF,
+  // These rules are nice and we want to configure over the defaults
   "@typescript-eslint/no-use-before-define": [
-    WARN,
+    "error",
     {
       functions: false,
       classes: false,
       variables: false,
       typedefs: false,
     },
   ],
-  "no-unused-expressions": OFF,
   "@typescript-eslint/no-unused-expressions": [
-    WARN,
+    "error",
     {
       allowShortCircuit: true,
       allowTernary: true,
       allowTaggedTemplates: true,
     },
   ],
-  "no-unused-vars": OFF,
   "@typescript-eslint/no-unused-vars": [
-    WARN,
+    "error",
     {
       args: "none",
       ignoreRestSiblings: true,
     },
   ],
-  "no-useless-constructor": OFF,
-  "@typescript-eslint/no-useless-constructor": WARN,
```

Comment:
This looks to only be include din the `strict` ruleset?

Reply:
Per https://typescript-eslint.io/rules/no-useless-constructor/#caveat it's got a bit too many false flags for us to feel comfortable enabling it. Do you want it despite the caveat?

Reply:
Nah happy to take the guidance there :)